site/content/contribute/desktop/debugging.md

```diff
@@ -6,16 +6,48 @@ date: 2019-01-22T00:00:00-05:00
 weight: 3
 ---
 
-The electron app itself can be inspected using the developer tools, available from the View menu:
+## Debugging in the Main Process
 
-![View toggle developer tools](/contribute/desktop/view-toggle-developer-tools.png)
+The simplest way to debug the main process is to simply insert logging statements wherever needed and have the application output logs of whatever is necessary.
 
-However, this view does not expose the contents of the webviews for inspection. To open the developer tools for a webview, first access the console for the electron shell as above, and then run
+If you'd like to make use of better debugging tools, you can use the Chrome Dev Tools or the debugger in VSCode by following the steps here: https://www.electronjs.org/docs/latest/tutorial/debugging-main-process
 
-```js
-document.getElementById("mattermostView0").openDevTools();
-```
+## Debugging in the Renderer Process
 
-Use `mattermostView0` to inspect the first opened tab, `mattermostView1` to inspect the second opened tab, and so on. A new window will appear with the developer tools for that webview:
+The renderer processes are controller by Chrome instances, so each of them will have their own Developer Tools instance.
+
+You can access these instances by going to the **View** menu (under the 3-dot menu on Windows/Linux, and in the top bar on macOS) and selecting:
+- **Developer Tools for Application Wrapper** for anything involving the top bar.
+- **Developer Tools for Current Tab** for anything involving the Mattermost view or the preload script
+    - **Note:** For this one, make sure you're currently on the tab where you want to load the Developer Tools. You can have instances open for tabs you aren't currently viewing, but to open them in the first place requires it to be opened.
+
+There are other `BrowserViews` that are governed seperately from the main application wrapper, including:
+- Dropdown Menu
+    - You can open this one by adding a line in the `main/teamDropdownView.ts` file. In the constructor, at the end, add:
+        ```js
+        this.view.webContents.openDevTools({mode: 'detach'});
+        ```
+- Modals
+    - You can open these by setting an environment variable when running the Desktop App called `MM_DEBUG_MODALS`.
+        ```
+        // macOS/Linux
+        export MM_DEBUG_MODALS=1
+
+        // Windows PowerShell
+        $env:MM_DEBUG_MODALS = 1
+        ```
+- URL View
+    - You can open this one by adding a line in the `main/viewManager.ts` file. In the function `showURLView`, at the end, add:
+        ```js
+        urlView.webContents.openDevTools({mode: 'detach'});
+        ```
+    - **Note:** This view is ephemeral and based on whether a link is hovered with the mouse, so it might be best to use some logging instead here.
+
+## Debugging the Mattermost Server/Webapp
```

Comment:
What is the best next step if the person is still uncertain about which Mattermost component the issue is in? I think I've seen elsewhere that the best place to report issues that someone is uncertain about to the Server repo. Is this the best solution here?

Reply:
Yeah I think that would make the most sense. Will add a bit for that :)